lib/gradebook.rb

```diff
@@ -1 +1,21 @@
+class GradeBook
+  attr_reader :instructor, :courses
 
+  def initialize(instructor)
+    @instructor = instructor
+    @courses = []
+  end
+
+  def add_course(course)
+    courses << course
+  end
+
+  def list_all_students
+    all_students = {}
+    @courses.each do |course|
+      all_students[course.name] = course.students   
```

Comment:
According to the interaction pattern this should be 
```suggestion
      all_students[course] = course.students   
```
So that the whole `course` object is the key, not just the name.